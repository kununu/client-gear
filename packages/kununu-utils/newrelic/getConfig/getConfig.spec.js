const isEnabled = require('../isEnabled');

const getConfig = require('./index');

jest.mock('../isEnabled', () => jest.fn());

describe('newRelic getConfig', () => {
  let installKey;
  let applicationName;

  beforeAll(() => {
    installKey = process.env.NR_INSTALL_KEY;
    applicationName = process.env.NR_APPLICATION_NAME;
    process.env.NR_INSTALL_KEY = 'mylicensekey';
    process.env.NR_APPLICATION_NAME = 'test_service;test_service_group';
  });

  afterAll(() => {
    process.env.NR_INSTALL_KEY = installKey;
    process.env.NR_APPLICATION_NAME = applicationName;
  });

  beforeEach(() => {
    isEnabled.mockClear();
  });

  it('returns the config including the supplied license id', () => {
    isEnabled.mockImplementation(() => true);
    expect(getConfig()).toMatchSnapshot();
  });

  it('returns the config without license id, if newRelic is disabled', () => {
    isEnabled.mockImplementation(() => false);
    expect(getConfig()).toMatchSnapshot();
  });
});
