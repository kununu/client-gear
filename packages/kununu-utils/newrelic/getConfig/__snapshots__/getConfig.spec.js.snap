// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newRelic getConfig returns the config including the supplied license id 1`] = `
Object {
  "allow_all_headers": false,
  "app_name": Array [
    "test_service",
  ],
  "attributes": Object {
    "exclude": Array [
      "request.headers.cookie",
      "request.headers.authorization",
      "request.headers.proxyAuthorization",
      "request.headers.setCookie*",
      "request.headers.x*",
      "response.headers.cookie",
      "response.headers.authorization",
      "response.headers.proxyAuthorization",
      "response.headers.setCookie*",
      "response.headers.x*",
    ],
  },
  "distributed_tracing": Object {
    "enabled": "true",
  },
  "license_key": "mylicensekey",
  "logging": Object {
    "level": "info",
  },
}
`;

exports[`newRelic getConfig returns the config without license id, if newRelic is disabled 1`] = `
Object {
  "allow_all_headers": false,
  "app_name": Array [
    "test_service",
  ],
  "attributes": Object {
    "exclude": Array [
      "request.headers.cookie",
      "request.headers.authorization",
      "request.headers.proxyAuthorization",
      "request.headers.setCookie*",
      "request.headers.x*",
      "response.headers.cookie",
      "response.headers.authorization",
      "response.headers.proxyAuthorization",
      "response.headers.setCookie*",
      "response.headers.x*",
    ],
  },
  "distributed_tracing": Object {
    "enabled": "true",
  },
  "license_key": "0000000000000000",
  "logging": Object {
    "level": "info",
  },
}
`;
